### DISCUSSION TITLE
Gitlab Automerge ignoreApprovals logic improvments

### DISCUSSION THREAD

**Discussion by fredrondina** [renovatebot/renovate#22096] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

gitlab.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

There is a problem with gitlab automerge ignoreApprovals logic.

Currently it makes a call to add a new rule with `approvals_required: 0`. 

`projects/${config.repository}/merge_requests/${pr}/approval_rules`

This new rule that is attempting to be created will be of gitlab rule type `any_approver`, of which gitlab only allows one rule of this type per project. If a project already has a rule of type `any_approver` this call fails with 

`
         "response": {
           "statusCode": 400,
           "statusMessage": "Bad Request",
           "body": {
             "message": {
               "rule_type": ["any-approver for the merge request already exists"]
             }
           },
`

I would like to put in a PR to update the logic in lib/modules/platform/gitlab/index.ts to first check for the presence of a rule of type `any_approver`, and if it exists update its `approvals_required` to 0 with a [PUT request](https://docs.gitlab.com/ee/api/merge_request_approvals.html#update-merge-request-level-rule) before proceeding with the current logic.

### What you tried so far.

looking at the logs and the code

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: POST https://gitlab.com/api/v4/projects/REDACTED/merge_requests/110/approval_rules = (code=ERR_NON_2XX_3XX_RESPONSE, statusCode=400 retryCount=0, duration=155) (repository=rREDACTED, branch=renovate/REDACTED-1.x)
       "trace_id": "5c528eb65f090284873705083674eddc",
       "span_id": "f9a13c5d204972df",
       "trace_flags": "01"
DEBUG: Gitlab API error (repository=REDACTED, branch=REDACTED-1.x)
       "err": {
         "name": "HTTPError",
         "code": "ERR_NON_2XX_3XX_RESPONSE",
         "timings": {
           "start": 1683807006495,
           "socket": 1683807006496,
           "lookup": 1683807006496,
           "connect": 1683807006500,
           "secureConnect": 1683807006504,
           "upload": 1683807006504,
           "response": 1683807006650,
           "end": 1683807006650,
           "phases": {
             "wait": 1,
             "dns": 0,
             "tcp": 4,
             "tls": 4,
             "request": 0,
             "firstByte": 146,
             "download": 0,
             "total": 155
           }
         },
         "message": "Response code 400 (Bad Request)",
         "stack": "HTTPError: Response code 400 (Bad Request)\n    at Request.<anonymous> (/opt/buildpack/tools/renovate/34.160.0/node_modules/got/dist/source/as-promise/index.js:118:42)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)",
         "options": {
           "headers": {
             "user-agent": "RenovateBot/34.160.0 (https://github.com/renovatebot/renovate)",
             "accept": "application/json",
             "authorization": "***********",
             "content-type": "application/json",
             "content-length": "57",
             "accept-encoding": "gzip, deflate, br",
             "traceparent": "00-5c528eb65f09028474eddc-2a0ad53f46c5bc4e-01"
           },
           "url": "https://gitlab.com/api/v4/projects/REDACTED/merge_requests/110/approval_rules",
           "hostType": "gitlab",
           "username": "",
           "password": "",
           "method": "POST",
           "http2": false
         },
         "response": {
           "statusCode": 400,
           "statusMessage": "Bad Request",
           "body": {
             "message": {
               "rule_type": ["any-approver for the merge request already exists"]
             }
```
</details>

**Comment by rarkins** [comment, marked as answer]
This appears to be the same as #22063 so best to move any new discussion there
